vmdb/app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -586,11 +589,9 @@ def prov_req_submit
   def prov_get_form_vars
     if params[:ids_checked]                         # User checked/unchecked a tree node
       ids = params[:ids_checked].split(",")
-      fld = @edit[:wf].kind_of?(MiqHostProvisionWorkflow) ? "tag_ids" : "vm_tags"
-      @edit[:new]["#{fld}".to_sym] = Array.new
-      ids.each do |id|
-        @edit[:new]["#{fld}".to_sym].push(id.to_i) if id != "" #for some reason if tree is not expanded clicking on radiobuttons this.getAllChecked() sends up extra blanks
-      end
+      @edit.store_path(:new, tag_symbol_for_workflow, [])
+      # for some reason if tree is not expanded clicking on radiobuttons this.getAllChecked() sends up extra blanks
+      @edit[:new][tag_symbol_for_workflow].push(ids.collect { |i| i.to_i unless i.blank? }.compact)
```

Comment:
Think this is pushing on an array and not the ids

```
[1,2,3].push(4) # => [1, 2, 3, 4]
[1,2,3].push([4,5,6]) # => [1, 2, 3, [4, 5, 6]]
```

Maybe:

```
@edit[:new][tag_symbol_for_workflow] += ids.collect { |i| i.to_i unless i.blank? }.compact
```
or
```
@edit[:new][tag_symbol_for_workflow] += ids.select(&:present?).collect(&:to_i)
```

(please double check my suggestions)